bundles/org.openhab.binding.bosesoundtouch/src/main/java/org/openhab/binding/bosesoundtouch/internal/handler/BoseSoundTouchHandler.java

```diff
@@ -517,32 +520,66 @@ public void handleGroupUpdated(BoseSoundTouchConfiguration masterPlayerConfigura
+                new ChannelTypeUID(BINDING_ID, CHANNEL_PRESET)).build());
+        allChannels.add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_BASS),
+                new ChannelTypeUID(BINDING_ID, CHANNEL_BASS)).build());
+        allChannels.add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_RATEENABLED),
+                new ChannelTypeUID(BINDING_ID, CHANNEL_RATEENABLED)).build());
+        allChannels.add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_SKIPENABLED),
+                new ChannelTypeUID(BINDING_ID, CHANNEL_SKIPENABLED)).build());
+        allChannels.add(
+                getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_SKIPPREVIOUSENABLED),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_SKIPPREVIOUSENABLED)).build());
+        allChannels.add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_SAVE_AS_PRESET),
+                new ChannelTypeUID(BINDING_ID, CHANNEL_SAVE_AS_PRESET)).build());
+        allChannels.add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_KEY_CODE),
+                new ChannelTypeUID(BINDING_ID, CHANNEL_KEY_CODE)).build());
+        allChannels
+                .add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_ALBUM),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_ALBUM)).build());
+        allChannels
+                .add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_ARTWORK),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_ARTWORK)).build());
+        allChannels
+                .add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_ARTIST),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_ARTIST)).build());
+        allChannels.add(
+                getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_DESCRIPTION),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_DESCRIPTION)).build());
+        allChannels
+                .add(getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_GENRE),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_GENRE)).build());
+        allChannels.add(
+                getCallback().createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_ITEMNAME),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_ITEMNAME)).build());
+        allChannels.add(getCallback()
+                .createChannelBuilder(new ChannelUID(getThing().getUID(), CHANNEL_NOWPLAYING_STATIONLOCATION),
+                        new ChannelTypeUID(BINDING_ID, CHANNEL_NOWPLAYING_STATIONLOCATION))
+                .build());

```

Comment:
Is there any formatting setting that would make this look better?

Reply:
I've fixed the part where it's sloppy code with a lot of unnecessary duplication causing all sorts of issues.